testgrid/cmd/gatherer/issuescraper.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gatherer
+
+import (
+	"encoding/json"
+	"github.com/sirupsen/logrus"
+	"io"
+	"k8s.io/test-infra/ghproxy/ghcache"
+	"k8s.io/test-infra/prow/github"
+	"net/http"
+)
+
+// JSONPresence detects if a particular key is present without actually parsing any of the data
+type JSONPresence struct {
+	present bool
+}
+
+func (j *JSONPresence) UnmarshalJSON([]byte) error {
+	j.present = true
+	return nil
+}
+
+type GitHubIssueScraper struct {
+	githubRoundTripper http.RoundTripper
+	RepoInfix          string
+}
+
+type GitHubIssue struct {
```

Comment:
I think you'd be able to import Prow's GitHub client and just use that, no implementation necessary